src/torrent/data/file_list.cc

```diff
@@ -592,6 +592,7 @@ FileList::create_chunk_part(FileList::iterator itr, uint64_t offset, uint32_t le
   // Check that offset != length of file.
 
   if (!(*itr)->prepare(prot))
+  if (!(*itr)->prepare(prot | ((*itr)->is_resize_queued() ? MemoryChunk::prot_write : 0)))
```

Comment:
Besides the duplicate if statement, adding `prot_write` is wrong.

open_file is always opened initially as read-only for a reason, it is not the stage where we verify or start changing files.

If truncation is required it needs to request write access explicitly at a higher level rather than depend on requesting write access on all calls to create chunk.

Reason why this was written as is was that we shouldn't truncate files that might have belonged to other downloads without explicit permission.